### PR TITLE
Fixed a bug with PreserveObjectReferences=true.

### LINKED ISSUE #10797: DCS_ArrayOfSimpleType_PreserveObjectReferences_True Failed
The following two tests throw `System.NotImplementedException` when using `SerializationOption.ReflectionOnly`,

DCS_ArrayOfSimpleType_PreserveObjectReferences_True
DCS_CircularTypes_PreserveObjectReferences_True

### PR DESCRIPTION
**Pull request by shmao**
Fixed a bug with PreserveObjectReferences=true. The issue happened when reflection based serialization is used. 

Fix #10797.

### CODE REVIEW

**Comment by shmao** [comment]
@mconnew @huanwu can you please review the PR? Thanks.

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionReader.cs [hunk 1]
```diff
@@ -221,36 +216,64 @@ protected object ReflectionReadValue(XmlReaderDelegator xmlReader, XmlObjectSeri
...
+                        throw new SerializationException(SR.Format(SR.ValueTypeCannotHaveId, DataContract.GetClrTypeFullName(type)));
+                    }
+
+                    if (primitiveContract != null && primitiveContract.UnderlyingType != Globals.TypeOfObject)
                     {
-                        objectId = context.GetObjectId();
-                        if (primitiveContract != null && primitiveContract.UnderlyingType != Globals.TypeOfObject)
-                        {
-                            value = primitiveContract.ReadXmlValue(xmlReader, context);
-                            context.AddNewObject(value);
```
Comment (shmao) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionReader.cs]: To reviewers: removing this, `context.AddNewObject(value)`, is essentially the fix. The ReadXmlValue above already adds the value object to the context. Adding the object again caused the issue.

**Comment by shmao** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test please

**Comment by shmao** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test please

There were xunit test failures in System.Net.Http.Functional.Tests

**Comment by mconnew** [comment]
LGTM

**Comment by shmao** [comment]
Thanks @mconnew for reviewing the PR. Merging it.

**Pull request merged**